### ISSUE TITLE
Changing uids and gids of user disrupts rootless podman setup

### ISSUE THREAD

**Issue by dnk8n** [containers/podman#8214]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

On a working rootless podman installation.

1. To assign a new UID to user called foo, enter:
`usermod -u 2005 foo`

2. To assign a new GID to group called foo, enter:
`groupmod -g 3000 foo`

3. `podman run ...`

**Describe the results you received:**

`Error: error creating tmpdir /tmp/run-1003/libpod/tmp: mkdir /tmp/run-1003/libpod: permission denied`

Note `1003` is the old uid

**Describe the results you expected:**

Podman should be able to regenerate the updated user information

**Additional information you deem important (e.g. issue happens only occasionally):**

If I manually chown /tmp/run-1003 to the new uid:gid, I get blank output (just hangs)

**Output of `podman version`:**

podman version 2.1.1

```
(paste your output here)
```

**Output of `podman info --debug`:**

Even `podman info --debug` command hangs. `podman ps` too.

```
(paste your output here)
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
Listing... Done
podman/unknown,now 2.1.1~2 amd64 [installed]
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Unsure

**Comment by mheon** [comment]
We don't presently have any way of adjusting directories in use after Podman has been run, because the paths that need to be changed spread rapidly throughout our database and it would be very difficult to find and change them all. In theory, `podman system migrate` could be extended to do this, so I'm going to retag this as an RFE to investigate a command to rewrite certain paths in the DB.

**Comment by dnk8n** [comment]
Ok great. For anyone else who might have stumbled into this, make sure for the moment that you don't plan to change uid/gids after already using podman!

I was able to simply create a new user with the correct uids/gids (needed due to data drive being mounted to two VMS, hence why I wanted the same user specs on both). I just deleted `user_a` and created `usera` (name changed for example) just in case recreating it wouldn't have worked.

Avoided the problem, but could be limiting in the future if `user_a` is off bounds in future.
